prov/usnic/src/usdf_fabric.c

```diff
@@ -341,6 +341,82 @@ static int validate_modebits(uint32_t version, const struct fi_info *hints,
 	return FI_SUCCESS;
 }
 
+static int usdf_alloc_fid_nic(struct fi_info *fi,
+			struct usd_device_attrs *dap)
+{
+	int ret;
+	struct fid_nic *nic = NULL;
+	struct fi_device_attr *da = NULL;
+	struct fi_bus_attr *ba = NULL;
+	struct fi_link_attr *la = NULL;
+
+	nic = ofi_nic_dup(NULL);
+	if (!nic)
+		goto nomem;
+
+	da = nic->device_attr;
+	da->name = strdup(dap->uda_devname);
+	if (!da->name)
+		goto nomem;
+	ret = asprintf(&da->device_id, "%s (%s)",
+		usd_devid_to_pid(dap->uda_vendor_id,
+				dap->uda_device_id),
+		usd_devid_to_nicname(dap->uda_vendor_id,
+				dap->uda_device_id));
+	if (ret < 0)
+		goto nomem;
+	ret = asprintf(&da->device_version, "0x%x", dap->uda_vendor_part_id);
+	if (ret < 0)
+		goto nomem;
+	ret = asprintf(&da->vendor_id, "0x%x", dap->uda_vendor_id);
+	if (ret < 0)
+		goto nomem;
+	da->driver = strdup("usnic_verbs");
+	if (!da->driver)
+		goto nomem;
+	da->firmware = strdup(dap->uda_firmware);
+	if (!da->firmware)
+		goto nomem;
+
+	ba = nic->bus_attr;
```

Comment:
Minor thing, but you probably don't need to use a variable for `ba` since you only use it once. Probably easier to just `nic->bus_addr->bus_type = FI_BUS_UNKNOWN`.